app/tests/test_search_endpoint.py

```diff
@@ -139,14 +107,51 @@ def test_query_multiple_non_allowed_parameters(self):
         """
         response = self.client.post(self.path, data=payload, content_type="application/json")
         self.assertStatusCode(400, response)
-        json_data_post = response.json()
+        json_data = response.json()
         for wrong_par in [wrong_query_parameter1, wrong_query_parameter2]:
             self.assertIn(
                 wrong_par,
-                str(json_data_post['description']),
+                str(json_data['description']),
                 msg=f"Wrong query parameter {wrong_par} not found in error message"
             )
 
+    def test_limit_in_post(self):
+        # limit in payload
+        limit = 1
+        payload = f"""
+        {{"query":
+            {{"created":
+                {{"lte":"9999-12-31T09:07:39.399892Z"}}
+            }},
+            "limit": {limit}
+        }}
+        """
+        response = self.client.post(self.path, data=payload, content_type="application/json")
+        self.assertStatusCode(200, response)
+        json_data_payload = response.json()
+        self.assertEqual(len(json_data_payload['features']), 1, msg="More than one item returned.")
+
+        # limit in url
+        payload = """
+        {"query":
+            {"created":
+                {"lte":"9999-12-31T09:07:39.399892Z"}
+            }
+        }
+        """
+        response = self.client.post(
+            f"{self.path}?limit={limit}", data=payload, content_type="application/json"
+        )
+        self.assertStatusCode(200, response)
+        json_data_url = response.json()
+
+        # compare limit in payload and limit in url
+        self.assertEqual(
+            json_data_payload['features'][0]['properties']['created'],
+            json_data_url['features'][0]['properties']['created'],
+            msg="Limit parameter in payload and in URL yielded different responses."
+        )
```

Comment:
This assert is not very strong and too hazardous (could be two different features that have been created at the same time). Both request above MUST returns exactly the same result ! So checking the whole response would be much better; `self.assertEqual(json_data_payload
, json_data_url)`. Also the goal of the test should be to check that both we can use `limit` either in payload or in query, however in this test we send it first only in payload and then both in payload and in query, I would in the second request only send limit in the query and not in the payload.

Reply:
Good point, I'll implement it like that

Reply:
I probably misunderstand you here. In the second test it is already implemented as you suggest. `Limit` is only in the URL and not in the payload (lines 134:141 don't contain a `limit`). Or what else do you mean?

Reply:
Oops my bad I didn't pay attention that the limit was not part anymore of the payload in the second request, I missed out the second payload definition in line 135. But this is dangerous, I would suggest to have the base payload in a variable without limit and using a dictionary instead of a string. Then for the first request you can make a copy of the payload and add the limit to it. I would also directly pass the dictionary to request and not a string with `json` attribute: ```
requests.post(url, json={'limit': 1, ...})
```
instead of 
```
requests.post(url, data='{"limit": 1}', content_type="application/json")
```
See https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests

Reply:
* `self.assertEqual(json_data_payload , json_data_url, msg=...)` would not work, as the `next links` and the `timestamps` would differ. I changed it to compare the item `id`. As it is unique, this assertion should do, I guess.
* using `json=...` in a requests.post does not work in the Django test client. But with `content_type=application/json` the `data=` will be serialized using `json.dumps()`.